deps/src/main/java/com/microsoft/azure/sdk/iot/deps/auth/IotHubCertificateManager.java

```diff
@@ -105,83 +110,60 @@
-            String line;
-            StringBuilder cert = new StringBuilder();
-            while ((line = certReader.readLine()) != null)
-            {
-                if (readLineCount > MAX_CERTIFICATE_LINES)
-                {
-                    throw new IllegalArgumentException("The provided certificate is too long");
-                }
-
-                cert.append(line);
-                cert.append("\r\n");
-                readLineCount++;
-            }
-
-            if (cert.length() == 0)
-            {
-                //Codes_SRS_IOTHUBCERTIFICATEMANAGER_25_004: [**This method shall throw IllegalArgumentException if certificate contents were empty.**]**
-                throw new IllegalArgumentException("Cert cannot be Empty");
-            }
+            certificates = certificateFactory.generateCertificates(fis);
+        }
 
-            this.validCert = cert.toString();
+        //Codes_SRS_IOTHUBCERTIFICATEMANAGER_34_008: [**If no certificates were parsed from the provided certificate file path, this function shall throw an IllegalArgumentException.**]**
+        if (certificates.isEmpty())
+        {
+            throw new IllegalArgumentException("Provided certificate path had no certificates");
         }
     }
 
     /**
      * Setter for the valid certificate to be used to communicate with IotHub
-     * @param cert valid certificate to be used to communicate with IotHub
+     * @param cert valid certificate to be used to communicate with IotHub. May be more than 1 certificate
      * @throws IllegalArgumentException If the cert string provided was null or empty
      */
-    void setValidCert(String cert) throws IllegalArgumentException
+    void setCertificates(String cert) throws IllegalArgumentException, IOException, CertificateException
     {
-        //Codes_SRS_IOTHUBCERTIFICATEMANAGER_25_009: [*If a user attempted to set the certificate and for somereason could not succeed then this method shall not use default certificate by setting valid certificate as null.**]**
-        this.validCert = null;
+        //Codes_SRS_IOTHUBCERTIFICATEMANAGER_34_006: [**This method shall throw IllegalArgumentException if parameter is null or empty.**]**
+        if (cert == null || cert.isEmpty())
+        {
+            throw new IllegalArgumentException("Certificate cannot be null or empty");
+        }
 
-        if (cert == null || cert.length() == 0)
+        //Codes_SRS_IOTHUBCERTIFICATEMANAGER_34_005: [**This function shall read the certificates from the provided
+        // string and save them into a certificate collection.**]**
+        try (InputStream inputStream = new ByteArrayInputStream(cert.getBytes()))
         {
-            //Codes_SRS_IOTHUBCERTIFICATEMANAGER_25_007: [**This method shall throw IllegalArgumentException if parameter is null.**]**
-            throw new IllegalArgumentException("Cert path cannot be null");
+            certificates = certificateFactory.generateCertificates(inputStream);
```

Comment:
One side effect here is if `cert` is empty, certificates will still be replaced and the instance is polluted. It might be alright since if someone want to customize certificates, they won't care about default value any more. IMPO, since there is exception thrown, I'd prefer certificates not be changed, validate and then set.